executor/slow_query.go

```diff
@@ -175,12 +226,12 @@ func ParseSlowLog(ctx sessionctx.Context, reader *bufio.Reader, fileLine, maxRow
 				}
 			} else if strings.HasSuffix(line, variable.SlowLogSQLSuffixStr) {
 				// Get the sql string, and mark the start flag to false.
-				_, err = st.setFieldValue(tz, variable.SlowLogQuerySQLStr, string(hack.Slice(line)), lineNum, checkValid)
+				_, err = st.setFieldValue(tz, variable.SlowLogQuerySQLStr, string(hack.Slice(line)), e.fileLine, e.checker)
 				if err != nil {
 					ctx.GetSessionVars().StmtCtx.AppendWarning(err)
 					continue
 				}
-				if checkValid == nil || checkValid(st.user) {
+				if e.checker == nil || e.checker.hasPrivilege(st.user) {
```

Comment:
It seems the `e.checker` always be assigned in the method `Initialize`, do we need to check it?

Reply:
In the unit test, the `e.checker` may be nil, but you are right, I will change the test and remove this check.